File: examples/write_file.go
package main

import (
	"github.com/kochavalabs/mazzaroth-xdr/xdr"
	"os"
)

const (
	xdrFile   = "xdrout"
	txWritten = 10000
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

// Write a transaction XDR file with txWritten TXs
func main() {
	signature := [64]byte{1, 2, 3}
	address := [32]byte{6, 7, 8}
	contract := []byte{1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}
	channelID := [32]byte{3, 4, 5}

	txXdr := xdr.Transaction{
		Signature: signature,
		Address:   address,
		Action: xdr.Action{
			ChannelID: channelID,
			Nonce:     4,
			Category: xdr.ActionCategory{
				Type: xdr.ActionCategoryTypeUPDATE,
				Update: &xdr.Update{
					Contract: contract,
				},
			},
		},
	}

	xF, err := os.Create(xdrFile)
	check(err)
	defer xF.Close()

	for i := 0; i < txWritten; i++ {
		xdr.Marshal(xF, txXdr)
	}
}

File: examples/write_file.js
const types = require('../js/xdr_generated.js').default
const xdr = require('xdr-js-serialize')
const fs = require('fs')

const tx = new types.Transaction()
const action = new types.Action() 
const update = new types.Update()
update.contract(Buffer.from('3a547668e859fb7b112a1e2dd7efcb739176ab8cfd1d9f224847fce362ebd99c', 'hex'))
action.channelId(Buffer.from('3a547668e859fb7b112a1e2dd7efcb739176ab8cfd1d9f224847fce362ebd99c', 'hex'))
action.nonce(xdr.UnsignedHyper.fromXDR([0, 0, 0, 0, 0, 0, 0, 0]))
action.category(types.ActionCategory.update(update))
tx.signature(Buffer.from('33a547668e859fb7b112a1e2dd7efcb739176ab8cfd1d9f224847fce362ebd99ca547668e859fb7b112a1e2dd7efcb739176ab8cfd1d9f224847fce362ebd99c', 'hex'))
tx.address(Buffer.from('3a547668e859fb7b112a1e2dd7efcb739176ab8cfd1d9f224847fce362ebd99c', 'hex'))
tx.action(action)


fs.open('./examples/xdroutjs', 'w', function(err, fd) {
    if (err) {
        throw 'could not open file: ' + err;
    }

  const toWrite = tx.toXDR()
  for (let i = 0; i < 10000; i++) {
    // write the contents of the buffer, from position 0 to the end, to the file descriptor returned in opening our file
    fs.write(fd, toWrite, 0, toWrite.length, null, function(err) {
        if (err) throw 'error writing file: ' + err;
    });
  }
});
